library/network/src/main/java/com/epmedu/animeal/network/NetworkStateProviderImpl.kt

```diff
@@ -0,0 +1,26 @@
+package com.epmedu.animeal.network
+
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+
+class NetworkStateProviderImpl(
+    private val connectivityManager: ConnectivityManager
+) : NetworkStateProvider {
+    override fun getWifiState(): Flow<NetworkState> {
```

Comment:
May be rename to getNetwork or smth similar? getWifi misleading according to WIFI, cellular transport